R/grouped_epi_archive.R

```diff
@@ -402,19 +337,17 @@ grouped_epi_archive =
                     .data_group_archive$DT = .data_group
                     comp_one_grp(.data_group_archive, .group_key, f = f, ...,
                                  ref_time_value = ref_time_value,
-                                 comp_effective_key_vars = comp_effective_key_vars,
```

Comment:
Great catch.  We don't have good test coverage for `epix_slide` with tidyeval, probably because it's hard to think up examples where it would be useful without access to the window data object; this will change with #313 merged.  We do have some (unrealistic/unuseful) example with tidy eval that could be repurposed as a test as well.

Reply:
TL;DR: Some debugging and rambling below.  There's an error message we eventually may want to improve.  Also, maybe relevant for #317 (impl or docs): we already are making a `.data` pronoun available which I forgot about, though it's not particularly useful, since it seems to have its own special class rather than `epi_df/tibble/`epi_archive`, doesn't support the functions we'd want to call on it, and we don't enable `pick`/etc.  I haven't had a chance to look at #317 yet, but documenting this seems like a pain.  Maybe we could try making `.data` the same as `.` and `.x`, though I'm not sure `rlang` will cooperate.

---

Ran into a sort of surprise, mostly because I accidentally mutated an archive and made the associated error messages more confusing.  We have to use `DT$time_value` or `.data$DT$time_value` to access the `time_value` column.  Without the mutation junk, the errors are semi-understandable:
```
Error in `map()`:
ℹ In index: 1.
Caused by error in `list2()`:
! object 'data_in' not found
Run `rlang::last_error()` to see where the error occurred.
```
or
```
Error in `map()`:
ℹ In index: 1.
Caused by error in `.data$data_in`:
! Column `data_in` not found in `.data`.
Run `rlang::last_error()` to see where the error occurred.
```
but the natural debug thing to do, trying to output `names(.data)`, doesn't work, as for some reason:
```{r}
library(rlang)
sloop::s3_get_method("names.rlang_data_pronoun")
#> function (x) 
#> {
#>     chr()
#> }
#> <bytecode: 0x5618d867d320>
#> <environment: namespace:rlang>
```

It seems we could provide some ways around this, e.g.:
```
DM = rlang::as_data_mask(ea$DT)
DM$.data <- ea
```
and feed that in as the `data` arg to `rlang::eval_tidy`.  But maybe the current behavior is the right way?  It'd be nice to have a clearer error message, though, but that seems complicated, and we might want to hold off until we consider alternative archive backends.

Reply:
- Found a hack to make current `.data` useful, only for `all_versions = TRUE`: `.data$clone()` will return an archive.
- And there's still no error!  Due to lazy evaluation + dots absorbing the erroneous argument.  (Our tidy eval `f` throws these dots away rather than trying to evaluate or forward.)

Reply:
Addressed this even though lazy eval saved us.